packages/web/src/components/comments/CommentBlock.tsx

```diff
@@ -35,21 +24,23 @@ export const CommentBlock = (props: CommentBlockProps) => {
   const {
     isPinned,
     message,
-    reactCount = 0,
-    trackTimestampS,
     id: commentId,
+    trackTimestampS,
     createdAt,
     userId: userIdStr
   } = comment
   const createdAtDate = useMemo(() => new Date(createdAt), [createdAt])
 
-  const { usePostComment, useDeleteComment, useReactToComment, usePinComment } =
-    useCurrentCommentSection()
+  const { usePostComment, useDeleteComment } = useCurrentCommentSection()
+
+  const [, { status: deleteStatus }] = useDeleteComment()
+  const prevDeleteStatus = usePrevious(deleteStatus)
+  useEffect(() => {
+    if (prevDeleteStatus !== deleteStatus && deleteStatus !== Status.LOADING) {
+      setIsDeleting(false)
+    }
+  }, [deleteStatus, prevDeleteStatus])
```

Comment:
I've done this quite a few times already, probably makes sense to update aquery to have some sort of callback like `onStatusChange` instead doing it manually like this 🤔

Reply:
Actually in hindsight the reason I have to do this is so that I can handle individual load states vs global ones (i.e. one comment delete status). Maybe this is just a side-effect of sharing the `useAQueryHook` in context vs using it at each component

Reply:
Would it make sense to write `useDeleteComment` as a standalone hook that reaches into the comment context, instead of sharing the hook itself via the comment context? Like this:

```ts
const useDeleteComment = () => {
  const { userId, entityId } = useCurrentCommentSection()
  const [deleteComment, deleteCommentResponse] = useDeleteCommentById()

  const wrappedHandler = async (commentId: string) => {
    if (userId) {
      deleteComment({ id: commentId, userId, entityId })
    }
  }
  return [wrappedHandler, deleteCommentResponse]
}
```

Reply:
yeah maybe this is the move. Let me see how this feels instead

Reply:
yea ok thats the way to go, I changed the delete to this and it worked as expected 👌

Reply:
Will change the others later on 💯 
Feel free to change them next week while I'm on call